flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/agg/AggsHandlerCodeGenerator.scala

```diff
@@ -866,10 +993,65 @@ class AggsHandlerCodeGenerator(
     """.stripMargin
   }
 
-  private def genEmitValue(): String = {
+  private def genEmitValue(isWindow: Boolean = false): String = {
     // validation check
     checkNeededMethods(needEmitValue = true)
-    aggBufferCodeGens(0).asInstanceOf[ImperativeAggCodeGen].emitValue
+    val methodName = "emitValue"
+    ctx.startNewLocalVariableStatement(methodName)
+
+    val windowCode =
+      if (isWindow) {
+        // no need to bind input
+        val exprGenerator = new ExprCodeGenerator(ctx, INPUT_NOT_NULL)
+        var valueExprs = Seq[GeneratedExpression]()
```

Comment:
Just use `val valueExprs = getWindowExpressions(windowProperties)`? Because `windowProperties` is empty when `hasNamespace` is false.